Scripts/Tracking/Query/ObscuranceQuery.cs

```diff
@@ -0,0 +1,91 @@
+﻿namespace VRTK.Core.Tracking.Query
+{
+    using System;
+    using System.Linq;
+    using UnityEngine;
+    using UnityEngine.Events;
+    using VRTK.Core.Cast;
+    using VRTK.Core.Process;
+
+    /// <summary>
+    /// Determines whether any <see cref="Collider"/> obscures a line between two <see cref="GameObject"/>s.
+    /// </summary>
+    /// <remarks>
+    /// The check is done by utilizing physics and as such a <see cref="Collider"/> is needed on <see cref="target"/>.
+    /// </remarks>
+    public class ObscuranceQuery : MonoBehaviour, IProcessable
+    {
+        /// <summary>
+        /// Defines the event with the <see cref="T:RaycastHit[]"/>.
+        /// </summary>
+        [Serializable]
+        public class HitEvent : UnityEvent<RaycastHit[]>
+        {
+        }
+
+        /// <summary>
+        /// The object from which a raycast will be done towards <see cref="target"/>.
+        /// </summary>
+        [Tooltip("The object from which a raycast will be done towards target.")]
+        public GameObject source;
+        /// <summary>
+        /// The object to which a raycast will be done from <see cref="source"/>.
+        /// </summary>
+        [Tooltip("The object to which a raycast will be done from source.")]
+        public Collider target;
+        /// <summary>
+        /// Optional settings to use when doing the raycast.
+        /// </summary>
+        [Tooltip("Optional settings to use when doing the raycast.")]
+        public PhysicsCast physicsCast;
+
+        /// <summary>
+        /// Emitted when another <see cref="Collider"/> exists on a line between <see cref="source"/> and <see cref="target"/>.
```

Comment:
Maybe 
`Emitted when the raycast from <see cref="source"/> is obscured from reaching <see cref="target"/> by another <see cref="Collider"/>`